circle/src/domain.rs

```diff
@@ -63,6 +63,12 @@ impl<F: ComplexExtendable> CircleDomain<F> {
         let coset1 = half_gen.shifted_powers(half_gen / self.shift);
         coset0.interleave(coset1).take(1 << self.log_n)
     }
+    pub(crate) fn nth_point(&self, idx: usize) -> Complex<F> {
```

Comment:
Might be cleaner if the input was a u64?

Reply:
hmm conceptually it's just "get the point at this index more efficiently than iterating the preceding points". So I think usize is fine. All callers want to call it with usize